js-arrays/arrays.js

```diff
@@ -0,0 +1,185 @@
+function isArrayOne(a) {
+  return a instanceof Array;
+}
+
+function isArrayTwo(a) {
+  return Array.isArray(a);
+}
+
+function isArrayThree(a) {
+  return Object.prototype.toString.call(a) === "[object Array]";
+}
+
+function range(a, b, c) {
+  var arr = [];
+  if (b === undefined) {
+    for (var i = 0; i < a; i++) {
+      arr.push(i);
+    }
+  } else if (c === undefined) {
+    for (var i = a; i < b; i++) {
+      arr.push(i);
+    }
+  } else if (c !== 0) {
+    for (var i = a; i < b; i += c) {
+      arr.push(i);
+    }
+  }
+  return arr;
+}
+
+function compactLoop(a) {
+  var arr = [];
+  if (Array.isArray(a)) {
+    for (var i = 0; i < a.length; i++) {
+      arr[i] = true;
+    }
+    return arr;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function compact(a) {
+  if (Array.isArray(a)) {
+    /*
+    var arr = a.map(function(b) {
+      return true;
+    });
+    */
+    var arr = a.slice();
+    return arr.fill(true);
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function sumLoop(a) {
+  if (Array.isArray(a)) {
+    var sum = 0;
+    for (var i = 0; i < a.length; i++) {
+      sum += a[i];
+    }
+    return sum;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function sum(a) {
+  if (Array.isArray(a)) {
+    var sum = 0;
+    a.map(function(b) {
```

Comment:
`map` returns new array, you can read more about it
in your case each should be used. but could you read about reduce?